kubeflow/trainer/rhai/transformers.py

```
@@ -110,6 +148,237 @@ def __post_init__(self):
+                    print("[Kubeflow] CUDA stream initialized for async checkpointing", flush=True)
+            except (ImportError, AttributeError):
+                print(
+                    "[Kubeflow] CUDA not available, checkpointing will be synchronous", flush=True
+                )
+
+        def setup_signal_handler(self):
+            """Register SIGTERM signal handler for JIT checkpointing."""
+            self._original_sigterm_handler = signal.signal(signal.SIGTERM, self._sigterm_handler)
+            print("[Kubeflow] JIT checkpoint signal handler registered for SIGTERM", flush=True)
+
+        def _sigterm_handler(self, signum, frame):
```

Ruff (0.14.5)

Unused method argument: `signum`

(ARG002)